vql/filesystem/filesystem.go

```diff
@@ -97,46 +93,88 @@ func (self GlobPlugin) Call(
 			}
 		}
 
-		globber := glob.NewGlobber().WithOptions(options)
+		organizedGlobs := make(map[string][]string)
+		var globalGlobs []string
 
-		// If root is not specified we try to find a common
-		// root from the globs.
+		// If root is not specified we organize the globs by their root and evaluate
```

Comment:
Can you explain why we need to refactor this code? The globber is independent from the accessors and i dont think will be affected by remapping.

Reply:
In general, I agree with you: the remapping does not affect the globber and how it uses the accessor(s). The refactoring was mostly needed because of the fact that the upstream code uses only one single accessor for all globs. While this works perfectly fine when there is only the host's filesystem to consider, it trips when the remapping is in place: depending on the path which is to be accessed, different accessors need to be used because of the remapping (or even all of them if a wildcard device or no specific device is selected). Therefore, the changeset introduces some code which organizes the globs by their device/root before the underlying data is accessed in the inline function.

My implementation is most probably not the best solution to this but it works to some degree and may serve as a starting point to build on - if we decide to dump it, be it so. In that case we will at least know that this is not the way the remapping may be implemented.